File: ChatCI/src/views/InterfaceUsuario.java
import java.util.Scanner;

public class InterfaceUsuario {
    private GerenciadorUsuarios gerenciador;

    public InterfaceUsuario(GerenciadorUsuarios gerenciador) {
        this.gerenciador = gerenciador;
    }

    public void exibirMenu() {
        Scanner scanner = new Scanner(System.in);
        while (true) {
            System.out.println("\n===== ChatCI - Menu Principal =====");
            System.out.println("1. Adicionar Usuário");
            System.out.println("2. Listar Usuários");
            System.out.println("3. Bloquear Usuário");
            System.out.println("4. Sair");
            System.out.print("Escolha uma opção: ");

            String opcao = scanner.nextLine();

            switch (opcao) {
                case "1":
                    adicionarUsuario(scanner);
                    break;
                case "2":
                    gerenciador.listarUsuarios();
                    break;
                case "3":
                    bloquearUsuario(scanner);
                    break;
                case "4":
                    System.out.println("Saindo...");
                    scanner.close();
                    return;
                default:
                    System.out.println("Opção inválida. Tente novamente.");
            }
        }
    }

    private void adicionarUsuario(Scanner scanner) {
        System.out.print("Nome: ");
        String nome = scanner.nextLine();
        System.out.print("Email: ");
        String email = scanner.nextLine();
        System.out.print("Tipo (Aluno, Professor, Administrador): ");
        String tipo = scanner.nextLine();
        gerenciador.adicionarUsuario(nome, email, tipo);
    }

    private void bloquearUsuario(Scanner scanner) {
        System.out.print("Digite o email do usuário a ser bloqueado: ");
        String email = scanner.nextLine();
        gerenciador.bloquearUsuario(email);
    }
}

File: ChatCI/views/interface_usuario.py
class InterfaceUsuario:
   def __init__(self, gerenciador):
      self.gerenciador = gerenciador

   def exibir_menu(self):
      while True:
         print("\n===== ChatCI - Menu Principal =====")
         print("1. Adicionar Usuário")
         print("2. Listar Usuários")
         print("3. Bloquear Usuário")
         print("4. Sair")
         opcao = input("Escolha uma opção: ")

         match opcao:
            case "1":
               self.adicionar_usuario()
            case "2":
               self.gerenciador.listar_usuarios()
            case "3":
               self.bloquear_usuario()
            case "4":
               print("Saindo...")
               break
            case _:
               print("Opção inválida. Tente novamente.")

   def adicionar_usuario(self):
      nome = input("Nome: ")
      email = input("Email: ")
      tipo = input("Tipo (Aluno, Professor, Administrador): ")
      self.gerenciador.adicionar_usuario(nome, email, tipo)

   def bloquear_usuario(self):
      email = input("Digite o email do usuário a ser bloqueado: ")
      self.gerenciador.bloquear_usuario(email)
